### ISSUE TITLE
Register validation error not forwarded

### ISSUE THREAD

**Issue by peteruithoven** [micky2be/superlogin-client#25]
It looks like `register()` tries to throw the `data` portion of an incoming error: https://github.com/micky2be/superlogin-client/blob/master/src/index.js#L297

Maybe something changed in Axios, because there isn't always a `data` property. Maybe this `data` property was moved to the `response` property. 

The contents of an error when giving a username that already exists, as it comes into the catch inside the register method.  

``` json
{
  "config": {
    "transformRequest": {},
    "transformResponse": {},
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/json;charset=utf-8"
    },
    "timeout": 0,
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
    "maxContentLength": -1,
    "method": "post",
    "url": "/auth/register",
    "data": "{\"username\":\"bob\",\"email\":\"bob@example.com\",\"password\":\"secret\",\"confirmPassword\":\"secret\"}"
  },
  "response": {
    "data": {
      "error": "Validation failed",
      "validationErrors": {
        "username": [
          "Username already in use"
        ]
      },
      "status": 400
    },
    "status": 400,
    "statusText": "Bad Request",
    "headers": {
      "date": "Wed, 12 Oct 2016 14:47:23 GMT",
      "etag": "W/\"66-LxtquFq4aEzEgxDmzU63QA\"",
      "connection": "keep-alive",
      "x-powered-by": "Express",
      "content-length": "102",
      "content-type": "application/json; charset=utf-8"
    },
    "config": {
      "transformRequest": {},
      "transformResponse": {},
      "headers": {
        "Accept": "application/json, text/plain, */*",
        "Content-Type": "application/json;charset=utf-8"
      },
      "timeout": 0,
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
      "maxContentLength": -1,
      "method": "post",
      "url": "/auth/register",
      "data": "{\"username\":\"bob\",\"email\":\"bob@example.com\",\"password\":\"secret\",\"confirmPassword\":\"secret\"}"
    },
    "request": {}
  }
}
```

Curiously, `login()` checks whether `data` is available and otherwise throws the original error: https://github.com/micky2be/superlogin-client/blob/master/src/index.js#L276-L281

**Comment by peteruithoven** [comment]
Same goes for the `register()` method, it's catch can receive the following error:

``` json
{
  "config": {
    "transformRequest": {},
    "transformResponse": {},
    "headers": {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/json;charset=utf-8"
    },
    "timeout": 0,
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
    "maxContentLength": -1,
    "method": "post",
    "url": "/auth/refresh",
    "data": "{}"
  },
  "response": {
    "data": "Unauthorized",
    "status": 401,
    "statusText": "Unauthorized",
    "headers": {
      "date": "Wed, 12 Oct 2016 15:04:24 GMT",
      "www-authenticate": "Bearer realm=\"Users\"",
      "connection": "keep-alive",
      "x-powered-by": "Express",
      "content-length": "12"
    },
    "config": {
      "transformRequest": {},
      "transformResponse": {},
      "headers": {
        "Accept": "application/json, text/plain, */*",
        "Content-Type": "application/json;charset=utf-8"
      },
      "timeout": 0,
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
      "maxContentLength": -1,
      "method": "post",
      "url": "/auth/refresh",
      "data": "{}"
    },
    "request": {}
  }
}
```

I also saw that at the following point we upgraded from Axios `^0.9.1` to `~0.14.0`;
https://github.com/micky2be/superlogin-client/commit/ed46f2b74753e7800d2e73194ec55d226aaec459

**Comment by micky2be** [comment]
Fix published with v0.4.1
